proposed/swift-interop.md

```diff
@@ -152,33 +152,40 @@ If possible, Swift tuples should be represented as `ValueTuple`s in .NET. If thi
 
 #### Projection Tooling Components
 
-The projection tooling should be split into these components:
+The projection tooling will be based on the [Binding Tools for Swift](https://github.com/xamarin/binding-tools-for-swift) (BTfS). The BTfS contains a set of tools that can consume a compiled Apple Swift library and generate wrappers (bindings) that allow it to be surfaced as a .NET library. This tool will be implemented as a self-hosted .NET CLI tool, automating the generation of wrappers by parsing the Swift library interface. It generates C# bindings that can be utilized as a .NET library and Swift bindings that C# bindings invoke in cases where direct P/Invoke into Swift library is not possible.
+
+The projection tooling should be split into these components.
 
 ##### Importing Swift into .NET
 
-1. A tool that takes in a `.swiftinterface` file or Swift sources and produces C# code.
-2. A library that provides the basic support for Swift interop that the generated code builds on.
-3. User tooling to easily generate Swift projections for a given set of `.framework`s.
-    - This tooling would build a higher-level interface on top of the tool in item 1 that is more user-friendly and project-system-integrated.
-4. (optional) A NuGet package, possibly referencable by `FrameworkReference` or automatically included when targeting macOS, Mac Catalyst, iOS, or tvOS platforms that exposes the platform APIs for each `.framework` that is exposed from Swift to .NET.
-    - This would be required to provide a single source of truth for Swift types so they can be exposed across an assembly boundary.
+Importing Swift into .NET is done through the following steps:
+
+1. The tool analyzes a source `.swift` library and maps entry points to mangled names.
+2. Then,the tool aggregates the public API; it takes in a `.swiftmodule` file (similar to a C header file) generated by the Swift compiler, creating a module declaration.
+3. For cases where direct P/Invoke from C# into Swift is not possible, the tool generates `.swift` source code wrappers along with a `.dylib` compiled native library.
+4. Subsequently, the tool aggregates the public API and generates module declarations for the generated wrappers.
```

Comment:
What is the module declaration that this is talking about? Is the module declaration just an internal implementation detail of the tool?

Reply:
It is an internal implementation detail of the tool. It is a representation with metadata of Swift APIs needed for wrappers and bindings generation.

Reply:
> an internal implementation detail of the tool

It would be better to avoid talking about it in this doc then.